src/php7/php_solr_client.c

```diff
@@ -1564,13 +1564,12 @@ PHP_METHOD(SolrClient, optimize)
 	xmlChar *request_string = NULL;
 	zend_bool success = 1;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "|sbb", &maxSegments, &maxSegmentsLen, &softCommit, &waitSearcher) == FAILURE) {
-
-		php_error_docref(NULL, E_WARNING, "Invalid parameter");
-
-		return;
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "|zbb", &maxSegmentsZval, &softCommit, &waitSearcher) == FAILURE) {
+		RETURN_THROWS();
 	}
 
+	SOLR_PARSE_PARAM_INT(maxSegmentsZval, maxSegments, maxSegmentsLen, "maxSegments");
```

Comment:
I don't like to change the signature here this drastically, unless we want to bump the major version.
If someone is using `optimize()` today in strict mode, then that code has been forced to provide a string `$maxSegments`. With this change we are throwing an exception when the extension is updated.

I would prefer we actually just change the `SOLR_PARSE_PARAM_INT` macro to allow strings in strict mode for now, and when we want do bump the major version, we can just change the macro to be properly strict :)

Reply:
sounds reasonable, I was just thinking about it